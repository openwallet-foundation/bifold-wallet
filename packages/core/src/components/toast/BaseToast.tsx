import React from 'react'
import { View, useWindowDimensions, StyleSheet, TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'

import { useTheme } from '../../contexts/theme'
import { GenericFn } from '../../types/fn'
import { testIdWithKey } from '../../utils/testable'
import Toast from 'react-native-toast-message'
import { ThemedText } from '../texts/ThemedText'

interface BaseToastProps {
  title?: string
  body?: string
  toastType: string
  onPress?: GenericFn
  onShow?: GenericFn
  onHide?: GenericFn
}

export enum ToastType {
  Success = 'success',
  Info = 'info',
  Warn = 'warn',
  Error = 'error',
}

const BaseToast: React.FC<BaseToastProps> = ({ title, body, toastType, onPress = () => null }) => {
  const { TextTheme, borderRadius, borderWidth, ColorPalette } = useTheme()
  const { width } = useWindowDimensions()
  const iconSize = 24
  let iconName = ''
  let backgroundColor = ''
  let borderColor = ''
  let iconColor = ''
  let textColor = ''
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'flex-start',
      flexDirection: 'row',
      marginTop: 25,
      borderWidth,
      borderRadius,
    },
    textContainer: {
      flexShrink: 1,
      marginVertical: 15,
      marginRight: 10,
    },
    icon: {
      marginTop: 15,
      marginHorizontal: 15,
    },
    title: {
      fontWeight: TextTheme.bold.fontWeight,
    },
    body: {
      marginTop: 10,
    },
  })
  switch (toastType) {
    case ToastType.Success:
      iconName = 'check-circle'
      backgroundColor = ColorPalette.notification.success
      borderColor = ColorPalette.notification.successBorder
      iconColor = ColorPalette.notification.successIcon
      textColor = ColorPalette.notification.successText
      break

    case ToastType.Info:
      iconName = 'info'
      backgroundColor = ColorPalette.notification.info
      borderColor = ColorPalette.notification.infoBorder
      iconColor = ColorPalette.notification.infoIcon
      textColor = ColorPalette.notification.infoText
      break

    case ToastType.Warn:
      iconName = 'report-problem'
      backgroundColor = ColorPalette.notification.warn
      borderColor = ColorPalette.notification.warnBorder
      iconColor = ColorPalette.notification.warnIcon
      textColor = ColorPalette.notification.warnText
      break

    case ToastType.Error:
      iconName = 'error'
      backgroundColor = ColorPalette.notification.error
      borderColor = ColorPalette.notification.errorBorder
      iconColor = ColorPalette.notification.errorIcon
      textColor = ColorPalette.notification.errorText
      break

    default:
      throw new Error('ToastType was not set correctly.')
  }

  return (
    <TouchableOpacity activeOpacity={1} onPress={() => onPress()}>
      <View style={[styles.container, { backgroundColor, borderColor, width: width - width * 0.1 }]}>
        <View style={{ flex: 1, flexDirection: 'row' }}>
          <Icon style={styles.icon} name={iconName} color={iconColor} size={iconSize} />
          <View style={styles.textContainer}>
            <ThemedText style={[styles.title, { color: textColor }]} testID={testIdWithKey('ToastTitle')}>
              {title}
            </ThemedText>
            {body && (
              <ThemedText style={[styles.body, { color: textColor }]} testID={testIdWithKey('ToastBody')}>
                {body}
              </ThemedText>
            )}
          </View>
        </View>
        <View>
          <TouchableOpacity
            onPress={() => {
              Toast.hide()
            }}
          >
            <Icon style={styles.icon} name={'close'} color={iconColor} size={iconSize} />
          </TouchableOpacity>
        </View>
      </View>
    </TouchableOpacity>
  )
}

export default BaseToast
