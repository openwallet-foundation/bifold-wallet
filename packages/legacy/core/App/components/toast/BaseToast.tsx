import React from 'react'
import { View, useWindowDimensions, StyleSheet, TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'

import { useTheme } from '../../contexts/theme'
import { GenericFn } from '../../types/fn'
import { testIdWithKey } from '../../utils/testable'
import Toast from 'react-native-toast-message'
import { ThemedText } from '../texts/ThemedText'

interface BaseToastProps {
  title?: string
  body?: string
  toastType: string
  onPress?: GenericFn
  onShow?: GenericFn
  onHide?: GenericFn
}

export enum ToastType {
  Success = 'success',
  Info = 'info',
  Warn = 'warn',
  Error = 'error',
}

const BaseToast: React.FC<BaseToastProps> = ({ title, body, toastType, onPress = () => null }) => {
  const { TextTheme, borderRadius, borderWidth, ColorPallet } = useTheme()
  const { width } = useWindowDimensions()
  const iconSize = 24
  let iconName = ''
  let backgroundColor = ''
  let borderColor = ''
  let iconColor = ''
  let textColor = ''
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'flex-start',
      flexDirection: 'row',
      marginTop: 25,
      borderWidth,
      borderRadius,
    },
    textContainer: {
      flexShrink: 1,
      marginVertical: 15,
      marginRight: 10,
    },
    icon: {
      marginTop: 15,
      marginHorizontal: 15,
    },
    title: {
      fontWeight: TextTheme.bold.fontWeight,
    },
    body: {
      marginTop: 10,
    },
  })
  switch (toastType) {
    case ToastType.Success:
      iconName = 'check-circle'
      backgroundColor = ColorPallet.notification.success
      borderColor = ColorPallet.notification.successBorder
      iconColor = ColorPallet.notification.successIcon
      textColor = ColorPallet.notification.successText
      break

    case ToastType.Info:
      iconName = 'info'
      backgroundColor = ColorPallet.notification.info
      borderColor = ColorPallet.notification.infoBorder
      iconColor = ColorPallet.notification.infoIcon
      textColor = ColorPallet.notification.infoText
      break

    case ToastType.Warn:
      iconName = 'report-problem'
      backgroundColor = ColorPallet.notification.warn
      borderColor = ColorPallet.notification.warnBorder
      iconColor = ColorPallet.notification.warnIcon
      textColor = ColorPallet.notification.warnText
      break

    case ToastType.Error:
      iconName = 'error'
      backgroundColor = ColorPallet.notification.error
      borderColor = ColorPallet.notification.errorBorder
      iconColor = ColorPallet.notification.errorIcon
      textColor = ColorPallet.notification.errorText
      break

    default:
      throw new Error('ToastType was not set correctly.')
  }

  return (
    <TouchableOpacity activeOpacity={1} onPress={() => onPress()}>
      <View style={[styles.container, { backgroundColor, borderColor, width: width - width * 0.1 }]}>
        <View style={{ flex: 1, flexDirection: 'row' }}>
          <Icon style={styles.icon} name={iconName} color={iconColor} size={iconSize} />
          <View style={styles.textContainer}>
            <ThemedText style={[styles.title, { color: textColor }]} testID={testIdWithKey('ToastTitle')}>
              {title}
            </ThemedText>
            {body && (
              <ThemedText style={[styles.body, { color: textColor }]} testID={testIdWithKey('ToastBody')}>
                {body}
              </ThemedText>
            )}
          </View>
        </View>
        <View>
          <TouchableOpacity
            onPress={() => {
              Toast.hide()
            }}
          >
            <Icon style={styles.icon} name={'close'} color={iconColor} size={iconSize} />
          </TouchableOpacity>
        </View>
      </View>
    </TouchableOpacity>
  )
}

export default BaseToast
