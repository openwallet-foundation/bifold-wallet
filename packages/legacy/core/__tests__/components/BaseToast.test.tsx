import { render } from '@testing-library/react-native'
import React from 'react'

import BaseToast, { ToastType } from '../../App/components/toast/BaseToast'

describe('BaseToast Component', () => {
  test('Info renders correctly', () => {
    const tree = render(
      <BaseToast
        title={'Hello World'}
        body={'The quick brown fox jumped over the lazy dog'}
        toastType={ToastType.Info}
      />
    )

    expect(tree).toMatchSnapshot()
  })

  test('Success renders correctly', () => {
    const tree = render(
      <BaseToast
        title={'Hello World'}
        body={'The quick brown fox jumped over the lazy dog'}
        toastType={ToastType.Success}
      />
    )

    expect(tree).toMatchSnapshot()
  })

  test('Warn renders correctly', () => {
    const tree = render(
      <BaseToast
        title={'Hello World'}
        body={'The quick brown fox jumped over the lazy dog'}
        toastType={ToastType.Warn}
      />
    )

    expect(tree).toMatchSnapshot()
  })

  test('Error renders correctly', () => {
    const tree = render(
      <BaseToast
        title={'Hello World'}
        body={'The quick brown fox jumped over the lazy dog'}
        toastType={ToastType.Error}
      />
    )

    expect(tree).toMatchSnapshot()
  })

  test('Toast Renders without body text', () => {
    const tree = render(<BaseToast title={'Hello World'} toastType={ToastType.Error} />)
    const bodyText = tree.queryByTestId('ToastBody')
    expect(bodyText).toBeNull()
  })
})
