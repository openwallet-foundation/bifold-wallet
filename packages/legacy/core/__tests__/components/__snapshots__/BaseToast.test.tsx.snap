// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseToast Component Error renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "borderRadius": 4,
          "borderWidth": 2,
          "flex": 1,
          "flexDirection": "row",
          "marginTop": 25,
        },
        Object {
          "backgroundColor": "#313132",
          "borderColor": "#D8292F",
          "width": 675,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#D8292F",
              "fontSize": 24,
            },
            Object {
              "marginHorizontal": 15,
              "marginTop": 15,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "flexShrink": 1,
            "marginRight": 10,
            "marginVertical": 15,
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Array [
                Object {
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#FFFFFF",
                },
              ],
            ]
          }
          testID="com.ariesbifold:id/ToastTitle"
        >
          Hello World
        </Text>
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Array [
                Object {
                  "marginTop": 10,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ],
            ]
          }
          testID="com.ariesbifold:id/ToastBody"
        >
          The quick brown fox jumped over the lazy dog
        </Text>
      </View>
    </View>
    <View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#D8292F",
                "fontSize": 24,
              },
              Object {
                "marginHorizontal": 15,
                "marginTop": 15,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`BaseToast Component Info renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "borderRadius": 4,
          "borderWidth": 2,
          "flex": 1,
          "flexDirection": "row",
          "marginTop": 25,
        },
        Object {
          "backgroundColor": "#313132",
          "borderColor": "#0099FF",
          "width": 675,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#0099FF",
              "fontSize": 24,
            },
            Object {
              "marginHorizontal": 15,
              "marginTop": 15,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "flexShrink": 1,
            "marginRight": 10,
            "marginVertical": 15,
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Array [
                Object {
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#FFFFFF",
                },
              ],
            ]
          }
          testID="com.ariesbifold:id/ToastTitle"
        >
          Hello World
        </Text>
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Array [
                Object {
                  "marginTop": 10,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ],
            ]
          }
          testID="com.ariesbifold:id/ToastBody"
        >
          The quick brown fox jumped over the lazy dog
        </Text>
      </View>
    </View>
    <View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#0099FF",
                "fontSize": 24,
              },
              Object {
                "marginHorizontal": 15,
                "marginTop": 15,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`BaseToast Component Success renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "borderRadius": 4,
          "borderWidth": 2,
          "flex": 1,
          "flexDirection": "row",
          "marginTop": 25,
        },
        Object {
          "backgroundColor": "#313132",
          "borderColor": "#2E8540",
          "width": 675,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#2E8540",
              "fontSize": 24,
            },
            Object {
              "marginHorizontal": 15,
              "marginTop": 15,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "flexShrink": 1,
            "marginRight": 10,
            "marginVertical": 15,
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Array [
                Object {
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#FFFFFF",
                },
              ],
            ]
          }
          testID="com.ariesbifold:id/ToastTitle"
        >
          Hello World
        </Text>
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Array [
                Object {
                  "marginTop": 10,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ],
            ]
          }
          testID="com.ariesbifold:id/ToastBody"
        >
          The quick brown fox jumped over the lazy dog
        </Text>
      </View>
    </View>
    <View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#2E8540",
                "fontSize": 24,
              },
              Object {
                "marginHorizontal": 15,
                "marginTop": 15,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`BaseToast Component Warn renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "borderRadius": 4,
          "borderWidth": 2,
          "flex": 1,
          "flexDirection": "row",
          "marginTop": 25,
        },
        Object {
          "backgroundColor": "#313132",
          "borderColor": "#FCBA19",
          "width": 675,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#FCBA19",
              "fontSize": 24,
            },
            Object {
              "marginHorizontal": 15,
              "marginTop": 15,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "flexShrink": 1,
            "marginRight": 10,
            "marginVertical": 15,
          }
        }
      >
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Array [
                Object {
                  "fontWeight": "bold",
                },
                Object {
                  "color": "#FFFFFF",
                },
              ],
            ]
          }
          testID="com.ariesbifold:id/ToastTitle"
        >
          Hello World
        </Text>
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Array [
                Object {
                  "marginTop": 10,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ],
            ]
          }
          testID="com.ariesbifold:id/ToastBody"
        >
          The quick brown fox jumped over the lazy dog
        </Text>
      </View>
    </View>
    <View>
      <View
        accessibilityState={
          Object {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          Object {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#FCBA19",
                "fontSize": 24,
              },
              Object {
                "marginHorizontal": 15,
                "marginTop": 15,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
